### PR TITLE
improvement: New inlay hints options

### PR DESCRIPTION
**Pull request by jkciesluk**
connected to https://github.com/scalameta/metals/pull/6236

### CODE REVIEW

File: packages/metals-vscode/package.json [hunk 1]
```diff
@@ -651,19 +653,29 @@
...
+        "title": "Toggle inlay hints for implicit parameters"
+      },
+      {
+        "command": "metals.toggle-inferred-types",
+        "category": "Metals",
+        "title": "Toggle inlay hints for inferred type"
       },
       {
-        "command": "metals.toggle-show-inferred-type",
+        "command": "metals.toggle-type-parameters",
```
Comment (tgodzik) [inline packages/metals-vscode/package.json:671]: Could we try to avoid rename the methods, it might break for people if they have shortcuts set up.

**Review by tgodzik — CHANGES_REQUESTED** [review]

File: packages/metals-vscode/package.json [hunk 2]
```diff
@@ -249,23 +249,25 @@
...
-        "metals.showInferredType": {
-          "type": "string",
-          "default": "false",
-          "enum": [
-            "false",
-            "true",
-            "minimal"
-          ],
-          "markdownDescription": "When this option is set to true inferred type in all possible places will be shown. This includes types for values, definitions, cases, type parameters. With 'minimal' option Metals will not show inferred type for type parameters and more complicated pattern matches."
+        "metals.inlayHints.inferredTypes.enable": {
```
[the remark this answers is not in the document]
Comment (jkciesluk) [reply packages/metals-vscode/package.json:252]: We don't need it, but this is how other languages do it (TypeScript, Rust, C++)

[the remark this answers is not in the document]
Comment (tgodzik) [reply packages/metals-vscode/package.json:252]: Ok, we can leave it then, though would be great to know the reason why they do it so

**Review by tgodzik — APPROVED** [review]
LGTM! When the PR in Metals is merged let's also bump snapshot version

**Review by tgodzik — APPROVED** [review]

**Pull request merged**